tests/unit/s2n_cipher_suite_match_test.c

```diff
@@ -781,6 +781,443 @@ int main(int argc, char **argv)
+                }
+                if (!s2n_chacha20_poly1305.is_available()) {
+                    EXPECT_FAILURE_WITH_ERRNO(s2n_set_cipher_as_tls_server(connection, dhe_chacha20_boosted_wire, count), S2N_ERR_CIPHER_NOT_SUPPORTED);
+                }
+            }
+
+            EXPECT_SUCCESS(s2n_config_free(rsa_only_config));
+        }
+
+        /* Test ChaCha20 boosting behaviour for tls 1.2 connection */
+        {
+            struct s2n_cipher_preferences cipher_preferences = { 0 };
+
+            struct s2n_security_policy security_policy = {
+                .minimum_protocol_version = S2N_TLS12,
+                .cipher_preferences = &cipher_preferences,
+                .kem_preferences = &kem_preferences_null,
+                .signature_preferences = &s2n_signature_preferences_20201021,
+                .ecc_preferences = &s2n_ecc_preferences_test_all,
+            };
+    
+            DEFER_CLEANUP(struct s2n_connection *connection = s2n_connection_new(S2N_SERVER), s2n_connection_ptr_free);
+            EXPECT_NOT_NULL(connection);
+
+            connection->security_policy_override = &security_policy;
+            connection->kex_params.server_ecc_evp_params.negotiated_curve = s2n_all_supported_curves_list[0];
+            connection->client_protocol_version = S2N_TLS12;
+            connection->actual_protocol_version = S2N_TLS12;
+            connection->server_protocol_version = S2N_TLS12;
+
+            EXPECT_SUCCESS(s2n_connection_set_config(connection, server_config));
+
+            if (s2n_chacha20_poly1305.is_available()) {
+                /* Chacha20 boosting support is enabled server-side with multiple potential chacha20 cipher suites */
+                static struct s2n_cipher_suite *cipher_suites_with_tls12_chacha20[] = {
+                    &s2n_ecdhe_ecdsa_with_aes_128_cbc_sha256,       /* 0xC0,0x23 */
+                    &s2n_ecdhe_ecdsa_with_chacha20_poly1305_sha256, /* 0xCC,0xA9 */
+                    &s2n_ecdhe_rsa_with_chacha20_poly1305_sha256,   /* 0xCC,0xA8 */
+                    &s2n_dhe_rsa_with_chacha20_poly1305_sha256,     /* 0xCC,0xAA */
+                };
+
+                cipher_preferences = (struct s2n_cipher_preferences) {
+                    .count = s2n_array_len(cipher_suites_with_tls12_chacha20),
+                    .suites = cipher_suites_with_tls12_chacha20,
+                    .allow_chacha20_boosting = true
+                };
+
+                /* Server and client have chacha20 boosting enabled. Server negotiates its most preferred
+                 * chacha20 cipher suite (which happens to also be the client's most preferred)
+                 */
+                {
+                    uint8_t ecdhe_chacha20_boosted_wire[] = {
+                        TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256,
+                        TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,
+                    };
```

Comment:
Again, wouldn't s2n_ecdhe_ecdsa_with_chacha20_poly1305_sha256 have been chosen even without chacha20 boosting? s2n_ecdhe_ecdsa_with_aes_128_cbc_sha256 isn't in the wire options.